packages/react/slot/src/Slot.tsx

```diff
@@ -11,20 +11,29 @@ interface SlotProps extends React.HTMLAttributes<HTMLElement> {
 
 const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
+  const childrenArray = React.Children.toArray(children);
+  const slottable = childrenArray.find(isSlottable);
+
+  if (slottable) {
+    const newElement = slottable.props.children;
+    const newChildren = childrenArray.map((child) => {
+      if (child === slottable) {
+        const slottableChildren = slottable.props.children as React.ReactNode;
+        if (React.Children.count(slottableChildren) > 1) return React.Children.only(null);
+        return React.isValidElement(slottableChildren)
+          ? (slottableChildren.props.children as React.ReactNode)
+          : null;
```

Comment:
Is it worth us providing some comments to explain the flow here? the children of children is a bit of a mind bender without context. Hard to know if I'm finding it easier to follow simply because I already have that context.

Reply:
I've added comment but also re-wrote it a bit which I think makes it easier to read.
Let me know your thoughts: https://github.com/radix-ui/primitives/pull/1376/commits/cb531c0c3faf85ae0e6efa4d872fcc4d9f88475d